sklearn/impute/_base.py

```diff
@@ -76,6 +76,11 @@ class _BaseImputer(TransformerMixin, BaseEstimator):
     It adds automatically support for `add_indicator`.
     """
 
+    _parameter_constraints = {
+        "missing_values": [numbers.Real, numbers.Integral, str, np.nan, None],
```

Comment:
`np.nan` is itself just a scalar, hence it is causing errors. Note that `isinstance(np.nan, Real)=True` so it's already covered by the lone `numbers.Real` constraint.
```suggestion
        "missing_values": [numbers.Real, numbers.Integral, str, None],
```

Reply:
> Not related to this, but I came across np.float32, but Real wont work for np.float32 and 64, but I guess the base is python float for both of them.

I'm not sure what you mean.
```python
import numpy as np
from numbers import Real

x = np.float32(1)
print(isinstance(x, Real)) # True

y = np.float64(1)
print(isinstance(y, Real)) # True
```

Reply:
Were you perhaps trying the following?

```python
x = np.float32
print(isinstance(x, Real)) # False
```